exam/helpers.py

```diff
@@ -155,8 +155,26 @@ def __init__(self, *calls):
     def __call__(self, *args, **kwargs):
         this_call = self.call_class(*args, **kwargs)
 
-        for call_obj, return_value in self:
-            if call_obj == this_call:
-                return return_value
+        for call, return_values in self:
+            if call == this_call:
+                try: 
+                    return next(return_values)
+                except StopIteration:
+                    #we're dealing with an empty iterator
+                    raise TypeError('Unknown effect for: %r, %r' % (args, kwargs))
+                except TypeError:
+                    #we're not dealing with an iterator
+                    if return_values.__class__.__name__ in ['list', 'tuple']:
+                        #only list or tuples should be turned into iterators, strings are excluded
```

Comment:
@Fluxx I'm thinking, perhaps we can be more liberal here and actually accept anything but dicts and strings? Something like:
`if not return_values.__class__.__name__ in ['dict', 'string']:`